### DISCUSSION TITLE
I have a problem in building tesr

### DISCUSSION THREAD

**Discussion by Yuanguo1994** [neperfepx/neper#109] [Q&A] [answered]
Hello, everyone

I am rookie in using the software. Now I encountered one problem that I failed in building raster tessellation in T direction. Could you please give me an example of building that?

**Comment by rquey** [comment, marked as answer]
Here is a simple command:
```
neper -T -n 10 -format tesr
```
Please be more specific if you need more information.

**Reply by Yuanguo1994** [reply]
Very thanks for your answer. I am reading the neper reference manual. It says that the using the raster tessellation could form the actual shape of grains as well as the grain orientation. In case I already have the form recoding the grain orientation and morphology, what kind command I should input to form such tessellation. Could you please give me a simple example?

**Reply by rquey** [reply]
Please clarify _(i)_ what is your input (experimental? 2D? 3D? statistical data? per-grain data?) and _(ii)_ what you would like to achieve: do you want to generate [a raster tessellation (`tesr`), a scalar tessellation (`tess`)](https://neper.info/generation.html#scalar-and-raster-formats), or [a mesh](https://neper.info/meshing.html#element-size)?

**Reply by Yuanguo1994** [reply]
Sorry for the too general questions. Actually, I want to input the grain orientation of per-grain into to a 3D raster tessellation.
And when I use the neper to mesh a tessellation, another question occurred. The following is my command and errors:


```
neper -M simulation.tess -order 2 -rcl 1.25 -part 2

Options `-part*' (mesh partitioning) are not available since this version of Neper has been built without libscotch.'
```

It seems something is missing. My neper version is 4.1.2. 

Could you tell me how to solve the problem?

By the way, I have run the command 'sudo apt install libscotch.' It showed that the libscothc is the newest version in computer.
 

regards

**Reply by rquey** [reply]
The easiest way to add the orientations to a tesr file is by editing the tesr file by hand. The file specifications are provided in the documentation.

To use the `-part*` options, you need to include libsctoch at configuration stage. This can be done using `cmake -DHAVE_LIBSCOTCH=ON ..` (or interactively, using `ccmake ..`). This should work out of the box if libscotch is installed in default system locations. Neper's header should then include something like
```
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
```
